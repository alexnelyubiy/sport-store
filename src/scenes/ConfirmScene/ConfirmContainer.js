import {
  compose, setDisplayName, withState, withHandlers, mapProps
} from 'recompose';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { resetCart } from 'actions';
import { getCartProducts, getUser, getPricesSum, getPrices } from 'selectors';
import ConfirmScene from './ConfirmScene';

export const handlers = {
  handleConfirm: props => () => {
    props.setConfirm(true);
  },
  handleResetCart: props => () => {
    props.dispatchResetCart();
  }
}

export const enhance = compose(
  setDisplayName('ProductsContainer'),
  connect(
    state => ({
      user: getUser(state),
      cartProducts: getCartProducts(state),
      pricesSum: getPricesSum(state),
      prices: getPrices(state)
    }), dispatch => bindActionCreators(
      {
        dispatchResetCart: resetCart,
      },
      dispatch,
    ),
  ),
  withState("isConfirm", "setConfirm", false),
  withHandlers(handlers),
  mapProps(props => ({
    ...props,
    user: props.user.toJS(),
    cartProducts: props.cartProducts.toJS(),
    prices: props.prices.toJS()
  })),
);

export default enhance(ConfirmScene);
