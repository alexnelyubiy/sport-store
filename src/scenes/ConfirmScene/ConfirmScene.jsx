import React from 'react';
import Master from 'scenes/Master';

const displayName = 'ConfirmScene';

function ConfirmScene() {
  return (
    <Master title="Confirmation">
      <div>
        Confirmation will be here!
      </div>
    </Master>
  );
}

ConfirmScene.displayName = displayName;

export default ConfirmScene;
