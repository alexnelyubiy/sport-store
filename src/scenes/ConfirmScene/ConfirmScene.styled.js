import styled, { keyframes } from "styled-components";

export const slideDown = keyframes`
  0% {
        transform: translateY(-800%);
    }
  50%{
      transform: translateY(10%);
  }
  65%{
      transform: translateY(-20%);
  }
  80%{
      transform: translateY(20%);
  }
  95%{
      transform: translateY(-10%);
  }           
  100% {
      transform: translateY(0%);
  }     
`;

export const ConfirmWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: auto 0;
  margin-bottom: 50px;
  overflow: hidden;
`;
export const ThanksWrapper = styled.div`
  height: 100%;
  display:flex;
  flex-direction: column;
  align-items: center;
`;
export const ThanksSpan = styled.span`
  margin-top: 70%;
  color: grey;
  font-size: 30px;
  animation-duration: 1.5s;
  animation-name: ${slideDown};
  visibility: visible !important;
  animation-timing-function: ease;
`;
