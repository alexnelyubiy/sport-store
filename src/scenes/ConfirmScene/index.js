import ConfirmScene from './ConfirmScene';

export default ConfirmScene;