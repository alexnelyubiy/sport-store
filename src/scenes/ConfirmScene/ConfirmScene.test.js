import React from 'react';
import { shallow } from 'enzyme';
import ConfirmScene from './ConfirmScene';

const testProps = {
  cartProducts: [{name: "Item 1", price: 290, id: 1}, {name: "Item 2", price: 20, id: 2}],
  pricesSum: 310,
  isCart: true,
  handleResetCart: jest.fn(),
  handleConfirm: jest.fn(),
  isConfirm: false,
  prices: {1: 290, 2: 20}
}
describe('Given the ConfirmScene component', () => {
  describe('when the component is rendered', () => {
    let component;
  
    beforeEach(() => {
			component = shallow(<ConfirmScene { ...testProps } />);
		});
  
    it('should render the component with provided props', () => {
      expect(component).toMatchSnapshot();
    });
    describe('and ConfirmScene is confirm', () => {
			beforeAll(() => {
				testProps.isConfirm = true;
			});

			it('should match the snapshot with emty cart', () => {
				expect(component).toMatchSnapshot();
			});
		});
  });
});
//

