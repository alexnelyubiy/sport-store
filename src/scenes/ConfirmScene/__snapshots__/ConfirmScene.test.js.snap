// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the ConfirmScene component when the component is rendered and ConfirmScene is confirm should match the snapshot with emty cart 1`] = `
<Master
  title="Confirmation"
>
  <ProductsScenestyled__ProductsWrapper>
    <ConfirmScenestyled__ThanksWrapper>
      <ConfirmScenestyled__ThanksSpan>
        Thanks for your choice!
      </ConfirmScenestyled__ThanksSpan>
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/"
      >
        <ConfirmButtonstyled__ConfirmButton
          onClick={[MockFunction]}
        >
          Back to Store
        </ConfirmButtonstyled__ConfirmButton>
      </NavLink>
    </ConfirmScenestyled__ThanksWrapper>
  </ProductsScenestyled__ProductsWrapper>
</Master>
`;

exports[`Given the ConfirmScene component when the component is rendered should render the component with provided props 1`] = `
<Master
  title="Confirmation"
>
  <ConfirmScenestyled__ConfirmWrapper>
    <ProductsScenestyled__ProductsWrapper>
      <ConfirmProduct
        key="1"
        product={
          Object {
            "id": 1,
            "name": "Item 1",
            "price": 290,
          }
        }
        totalPrice={290}
      />
      <ConfirmProduct
        key="2"
        product={
          Object {
            "id": 2,
            "name": "Item 2",
            "price": 20,
          }
        }
        totalPrice={20}
      />
    </ProductsScenestyled__ProductsWrapper>
    <CardScenestyled__TotalWrap>
      <span>
        Total price: 
        310
        $
      </span>
    </CardScenestyled__TotalWrap>
    <UserConfirm />
    <ConfirmButtonstyled__ConfirmButton
      onClick={[MockFunction]}
    >
      Confirm
    </ConfirmButtonstyled__ConfirmButton>
  </ConfirmScenestyled__ConfirmWrapper>
</Master>
`;
